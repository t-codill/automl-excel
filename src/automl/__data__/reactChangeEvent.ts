// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
