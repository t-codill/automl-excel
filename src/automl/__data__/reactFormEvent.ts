// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
