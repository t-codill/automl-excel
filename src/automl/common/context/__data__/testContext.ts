import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
