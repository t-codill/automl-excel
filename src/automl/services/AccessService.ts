import { AuthorizationManagementClient, AuthorizationManagementModels } from "@azure/arm-authorization";
import { includes, map, some } from "lodash";
import { IServiceBaseProps } from "./ServiceBase";
import { ServiceBaseArm } from "./ServiceBaseArm";

function hasPermission(permissionActions: Array<string[] | undefined>): boolean {
    return some(permissionActions, (permissionAction) => includes(permissionAction, "*"));
}

export class AccessService extends ServiceBaseArm<AuthorizationManagementClient> {
    constructor(props: IServiceBaseProps) {
        super(props, AuthorizationManagementClient);
    }

    public async getAccess(): Promise<AuthorizationManagementModels.PermissionGetResult | undefined> {
        return this.send(async (client, abortSignal) => {
            return client.permissions.listForResource(
                this.props.resourceGroupName,
                "Microsoft.MachineLearningServices",
                "",
                "workspaces",
                this.props.workspaceName, {
                    abortSignal
                });
        });
    }
    public readonly checkPermission = async () => {
        const curPermissions = await this.getAccess();
        if (!curPermissions) {
            return undefined;
        }
        const permissionActions = map(curPermissions, (permission) => permission && permission.actions);
        return hasPermission(permissionActions);
    }
}
