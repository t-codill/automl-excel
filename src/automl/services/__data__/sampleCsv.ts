export const sampleCsv = `a,b,c,d,e
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
`;

const largeCsvHeader = Array(50)
    .fill("aaaaa,bbbbb,ccccc,ddddd,eeeee")
    .join(",");

const largeCsvRow = Array(50)
    .fill("11111,22222,33333,44444,55555")
    .join(",");

export const largeCsv = `${largeCsvHeader}
${Array(50)
        .fill(largeCsvRow)
        .join("\n")}
`;
