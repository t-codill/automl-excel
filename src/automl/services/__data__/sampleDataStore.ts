import { AzureMachineLearningDatastoreManagementModels } from "@vienna/datastore";

export const sampleDataStore: Required<AzureMachineLearningDatastoreManagementModels.DataStoreDto> = {
    dataStoreType: "AzureBlob",
    name: "defaultDataStore",
    azureStorageSection: {
        accountKey: "key",
        accountName: "account",
        credentialType: "AccountKey",
        credential: "secret",
        containerName: "defaultContainer",
        isSas: false
    },
    azureDataLakeSection: {},
    azurePostgreSqlSection: {},
    azureSqlDatabaseSection: {},
    hasBeenValidated: true
};
