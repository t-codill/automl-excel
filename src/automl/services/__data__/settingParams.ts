import { AdvancedSetting } from "../AdvancedSetting";

export const settingParams: AdvancedSetting = {
    jobType: "classification",
    column: "1",
    metric: "AUC_weighted",
    experimentTimeoutMinutes: 60,
    maxIteration: 15,
    experimentExitScore: 0.99,
    maxConcurrent: 1,
    maxCores: 2,
    preprocess: true,
    nCrossValidations: 5,
    blacklistAlgos: null,
    validationSize: 0.2,
    grainColumns: null,
    maxHorizon: null,
    timeSeriesColumn: null
};
