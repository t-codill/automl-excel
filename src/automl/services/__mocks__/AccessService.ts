import { AuthorizationManagementModels } from "@azure/arm-authorization";

export class AccessService {

    public async getAccess(): Promise<AuthorizationManagementModels.PermissionGetResult | undefined> {
        return [{
            actions: ["*"]
        }];
    }

    public async checkPermission(): Promise<boolean | undefined> {
        return true;
    }

    public dispose(): void {
        return;
    }

    public reset(): void {
        return;
    }
}
