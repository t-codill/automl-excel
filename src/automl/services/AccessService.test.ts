import { AuthorizationManagementClient } from "@azure/arm-authorization";
import { restCanceledError } from "../../__data__/restCanceledError";
import { testContext } from "../common/context/__data__/testContext";
import { AccessService } from "./AccessService";

jest.mock("@azure/arm-authorization");

const service = new AccessService(testContext);

describe("AccessService", () => {
    let listForResourceSpy: jest.SpyInstance;
    beforeEach(() => {
        listForResourceSpy = jest.spyOn(AuthorizationManagementClient.prototype.permissions, "listForResource");
    });
    it("should getAccess", async () => {
        const result = await service.getAccess();
        expect(result)
            .toMatchSnapshot();
    });

    it("should return undefined if permissions is undefined", async () => {
        listForResourceSpy.mockImplementation(() => {
            throw restCanceledError;
        });
        expect(await service.checkPermission())
            .toBeUndefined();
    });
    it("should return false if permission is for reader", async () => {
        listForResourceSpy.mockReturnValue(Promise.resolve([
            {
                actions: [
                    "*/read"
                ],
                notActions: [],
                dataActions: [],
                notDataActions: []
            }]));
        expect(await service.checkPermission())
            .toEqual(false);
    });

    it("should return true if permission is for owner or contributor", async () => {
        listForResourceSpy.mockReturnValue(Promise.resolve([
            {
                actions: [
                    "*"
                ],
                notActions: [],
                dataActions: [],
                notDataActions: []
            }]));
        expect(await service.checkPermission())
            .toEqual(true);
    });
});
