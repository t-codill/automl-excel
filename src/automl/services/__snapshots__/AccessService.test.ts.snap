// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessService should getAccess 1`] = `
Array [
  Object {
    "actions": Array [
      "*",
    ],
    "dataActions": Array [],
    "notActions": Array [
      "Microsoft.Authorization/*/Delete",
      "Microsoft.Authorization/*/Write",
      "Microsoft.Authorization/elevateAccess/Action",
      "Microsoft.Blueprint/blueprintAssignments/write",
      "Microsoft.Blueprint/blueprintAssignments/delete",
    ],
    "notDataActions": Array [],
  },
  Object {
    "actions": Array [
      "*",
    ],
    "dataActions": Array [],
    "notActions": Array [],
    "notDataActions": Array [],
  },
  Object {
    "actions": Array [
      "*/read",
    ],
    "dataActions": Array [],
    "notActions": Array [],
    "notDataActions": Array [],
  },
]
`;
